.github/workflows/action-on-PR-labeled.yml

```diff
@@ -1,22 +1,23 @@
-name: Action on PR labeled
-
-# This workflow has write permissions on the repo
-# It must not checkout a PR and run untrusted code
+name: On labeled PR
 
 on:
   pull_request_target:
-    types:
-      - labeled
+    types: [labeled]
 
+# This workflow can modify pull requests.
+# It must not checkout a PR and run untrusted code.
```

Comment:
this comment feels "wronger" than it was before. here the `permissions` key is actually restricting permissions to only PR, but by default it has write permissions on the repo

See the warning at https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

Reply:
I saw this note in the [`permissions`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions) documentation:

> "For each of the available scopes, shown in the table below, you can assign one of the permissions: read, write, or none. If you specify the access for any of these scopes, all of those that are not specified are set to none."

From the [`pull_requests`](https://docs.github.com/en/rest/authentication/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-pull-requests) API docs, it appears to only have access to modifying and deleting pull requests.

I read the warning you linked, but wouldn't the only harm be deleting pull requests, comments, and milestones? How do you suggest modifying the comment?

Reply:
I would suggest not modifying it?

This workflow has a token with write permissions on the whole repo.

The `permissions` part restricts the permissions only to PRs, you could keep the existing comment unmodified and add a new one on the `permissions` saying that it must not change

Reply:
Ok, I fixed it in 9a527dbb6dcba447abd79a9e2c670ed39d84b75e by returning the original message and converting my comment to a simple explanation on why the workflow needs additional access to pull requests.